runtime/backend/options.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+#include <executorch/runtime/core/error.h>
+#include <executorch/runtime/core/span.h>
+#include <cstddef>
+#include <cstring>
+#include <variant>
+
+namespace executorch {
+namespace runtime {
+
+// Strongly-typed option key template
+template <typename T>
+struct OptionKey {
+  const char* key;
+  constexpr explicit OptionKey(const char* k) : key(k) {}
+};
+
+// Union replaced with std::variant
+using OptionValue = std::variant<bool, int, const char*>;
+
```

Comment:
Document the safety and lifecycle of string types

All string keys and values must have static storage duration (string literals, static const char arrays, or global constants). The BackendOptions class does NOT take ownership of strings.

Reply:
For the actual value itself, I think we should do this where std::variant itself holds array of chars rather than pointer to char. Like this https://godbolt.org/z/odTc9ex96